src/time.rs

```diff
@@ -0,0 +1,170 @@
+//! Time structures.
+//!
+//! The `time` modules contains structures used to represent both
+//! absolute and relative time.
+
+use core::{ops, fmt};
+
+static MILLIS_PER_SEC: u64 = 1_000;
+
+/// A representation of an absolute time value.
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Instant {
+    millis: u64,
+}
+
+impl Instant {
+    /// Create a new `Instant`.
+    pub fn new(millis: u64) -> Instant {
+        Instant { millis }
+    }
+
+    /// Add a [Duration] to the current timestamp value and return a
+    /// `Option` containing a new [Instant] of the result on success.
+    ///
+    /// [Duration]: struct.Duration.html
+    /// [Instant]: struct.Instant.html
+    pub fn checked_add(&self, rhs: Duration) -> Option<Instant> {
+        self.millis.checked_add(rhs.millis()).map(|x| Instant::new(x))
+    }
+
+    /// Subtract a [Duration] from the current timestamp value and
+    /// return a `Option` containing a new [Instant] of the result on
+    /// success.
+    ///
+    /// [Duration]: struct.Duration.html
+    /// [Instant]: struct.Instant.html
+    pub fn checked_sub(&self, rhs: Duration) -> Option<Instant> {
+        self.millis.checked_sub(rhs.millis()).map(|x| Instant::new(x))
+    }
+
+    /// Perform a saturating subtraction with a [Duration] from
+    /// the current timestamp value and return a new [Instant] of
+    /// the result.
+    ///
+    /// [Duration]: struct.Duration.html
+    /// [Instant]: struct.Instant.html
+    pub fn saturating_sub(&self, interval: Duration) -> Instant {
+        Instant::new(self.millis.saturating_sub(interval.millis()))
+    }
+
+
+    /// The number of milliseconds that have passed since an arbitrary
+    /// point in time.
```

Comment:
"an arbitrary point in time" → "the beginning of time", and the docstring for `Instant` itself should explain that the zero value of `Instant` is inherently arbitrary.